### DISCUSSION TITLE
Gitpod integration

### DISCUSSION THREAD

**Discussion by 0Grit** [xpack/vscode-xpack-extension-ts#26] [Show and tell]
Working on getting the xpack extension up and running with the xpack hello world template in gitpod.io
Will share my progress here. 

gitpod.io/#https://github.com/0Grit/hello-world-template-xpack/tree/0grit-add-gitpod-support

**Comment by 0Grit** [comment]
I have the latest Microsoft C/C++ extension installed. 
![image](https://user-images.githubusercontent.com/24318071/143443250-95b3004b-1038-4301-9ea2-979e20288282.png)


Getting the following error.
![image](https://user-images.githubusercontent.com/24318071/143443119-3aff7106-eff8-4814-9d59-48d4a2ca6f02.png)

**Reply by 0Grit** [reply]
I am manually able to open a view

![image](https://user-images.githubusercontent.com/24318071/143443926-ca285485-f8f6-411d-a59d-8c5d88545a4f.png)

**Reply by 0Grit** [reply]
Maybe due to not having xpm installed? Going to add that to the gitpod pre-build now.

**Reply by 0Grit** [reply]
No change in behavior still need to enable the view manually.

**Comment by 0Grit** [comment]
Running `xpm install`
![image](https://user-images.githubusercontent.com/24318071/143445060-d24f1451-50b5-49c1-a3dc-f300d219dcf3.png)

**Reply by 0Grit** [reply]
I am cloning https://github.com/0Grit/hello-world-template-xpack/tree/0grit-add-gitpod-support versus using `xpm init`

**Reply by ilg-ul** [reply]
I entered a bug report for the install issue (https://github.com/xpack/xpm-js/issues/119).

**Comment by ilg-ul** [comment]
I'm kind of lost in what you are trying to achieve... 

I retain the issue with `xpm install` in the template project. Try `npm install`.

**Reply by 0Grit** [reply]
Hard to explain will need to show once i have a better feel for xpm.

**Reply by ilg-ul** [reply]
Perhaps we should talk, to help you understand it better than via the documentation pages (which, I agree, are not great).

**Reply by 0Grit** [reply]
https://blog.golioth.io/we-tried-to-debug-a-microcontroller-from-our-browser/

Probably won't be able to talk today but open to do so Friday or maybe next week. I'll send an email.

Probably pausing for the day now that the rest of the household is awake and today is a US Holiday.

**Reply by ilg-ul** [reply]
Ok, I'm GMT+2, but if we arrange by email, probably we can find a common time.

**Comment by ilg-ul** [comment]
As for the _debug a microcontroller from a browser_, I'd not be surprised to find various hacks, but I'm stuck in my stupidity, I don't understand why is this better than debugging locally. :-(

**Reply by 0Grit** [reply]
Its not better than debugging locally but if webusb and probe-rs are paired with gitpod and xpacks a user could build, flash, and debug in less than 2 minutes. They would have 0 dependencies to install on their local machine besides a browser.

**Reply by ilg-ul** [reply]
For a simple blinky demos it might be ok, but for serious projects I don't think that remote debugging is a joy. 

How does SEGGER RTT run over this setup?

**Reply by 0Grit** [reply]
Probably not well. Regular debug was laggy, I did not try Segger's network debug server thing which is "optimized" for the internet.

Until someone comes out with a MS "Debug Adapter Protocol" debugger that runs natively on the client side of the browser  and uses webusb It will not be a very pleasant experience.

**Reply by 0Grit** [reply]
For now I am mainly running QEMU targets in gitpod versus using it with real hardware.

**Reply by ilg-ul** [reply]
For unit tests QEMU is ok, but for real projects I doubt you can get very far.
